apps/prairielearn/src/models/course-request.ts

```diff
@@ -0,0 +1,99 @@
+import { loadSqlEquiv, queryRow, queryAsync } from '@prairielearn/postgres';
+import { z } from 'zod';
+import { logger } from '@prairielearn/logger';
+import * as Sentry from '@prairielearn/sentry';
+
+import { InstitutionSchema, DateFromISOString, IdSchema } from '../lib/db-types';
+import { createCourseRepoJob } from '../lib/github';
+import { sendCourseRequestMessage } from '../lib/opsbot';
+
+const sql = loadSqlEquiv(__filename);
+
+const JobsRowSchema = z.object({
+  authn_user_id: IdSchema.optional(),
+  finish_date: DateFromISOString.optional(),
+  id: IdSchema.optional(),
+  number: z.number().optional(),
+  start_date: DateFromISOString.optional(),
+  status: z.string().optional(),
+});
+
+const CourseRequestRowSchema = z.object({
+  approved_by_name: z.string().nullable(),
+  first_name: z.string().nullable(),
+  github_user: z.string().nullable(),
+  id: IdSchema,
+  institution: z.string().nullable(),
+  jobs: JobsRowSchema,
+  last_name: z.string().nullable(),
+  short_name: z.string().nullable(),
+  status: z.enum(['pending', 'approved', 'denied', 'creating', 'failed']),
+  title: z.string().nullable(),
+  user_id: IdSchema.nullable(),
+  user_name: z.string().nullable(),
+  work_email: z.string().nullable(),
+});
+
+export async function getCourseRequests(show_all: boolean) {
+  return await queryRow(
+    sql.get_requests,
+    { show_all },
+    z.object({
+      institutions: z.array(InstitutionSchema),
```

Comment:
This function shouldn't be concerned with fetching institutions; this kind of thing would better belong in `models/institution.ts`.

Note that this query is technically returning a combination of institution and its enabled auth providers, which the schema here doesn't capture. That said: I can't actually see where the list of institutions is used on either the admin courses or admin course requests pages. Consider checking if you need to query for them at all!

Reply:
Unfortunately, we are using the institutions data for the dropdown in the approve course request form. I have moved this piece to `models/institution.ts` and added a query to select all institutions in `models/institution.sql`.

Reply:
Ah, right, I forgot about that dropdown. Thanks for splitting out the query!